File: src/main/java/fr/epyi/metropiamod/commands/ClearSkinCacheCommand.java
package fr.epyi.metropiamod.commands;

import com.mojang.brigadier.Command;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.exceptions.CommandSyntaxException;
import fr.epyi.metropiamod.CustomSkinManager;
import fr.epyi.metropiamod.MetropiaMod;
import fr.epyi.metropiamod.network.PacketHandler;
import fr.epyi.metropiamod.network.client.ClientClearSkinCache;
import net.minecraft.command.CommandSource;
import net.minecraft.util.text.StringTextComponent;
import net.minecraft.util.text.TextComponent;
import net.minecraft.util.text.TranslationTextComponent;

import static net.minecraft.command.Commands.literal;

public class ClearSkinCacheCommand {

    public static void register(CommandDispatcher<CommandSource> dispatcher) {

        // Thing to note, arguments are handled in alphabetical order.
        LiteralArgumentBuilder<CommandSource> clear = literal("reloadskin").executes(ClearSkinCacheCommand::run);

        dispatcher.register(clear);
    }

    private static int run(CommandContext<CommandSource> ctx) {
        try {
            PacketHandler.sendToPlayer(new ClientClearSkinCache(), ctx.getSource().asPlayer());
            CustomSkinManager.sendAllToPlayer(ctx.getSource().asPlayer(), false);
        } catch (CommandSyntaxException e) {
            MetropiaMod.LOGGER.info("This command can only be run from the console");
        }
        ctx.getSource().sendFeedback(new TranslationTextComponent("messages.reloadSkinSuccess"), false);
        return Command.SINGLE_SUCCESS;
    }
}

File: src/main/java/fr/epyi/metropiamod/commands/ClearSkinCacheCommand.kt
package fr.epyi.metropiamod.commands

import com.mojang.brigadier.Command
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import com.mojang.brigadier.exceptions.CommandSyntaxException
import fr.epyi.metropiamod.CustomSkinManager
import fr.epyi.metropiamod.network.PacketHandler
import fr.epyi.metropiamod.network.client.ClientClearSkinCache
import net.minecraft.command.CommandSource
import net.minecraft.command.Commands
import net.minecraft.util.text.TranslationTextComponent

object ClearSkinCacheCommand {
    fun register(dispatcher: CommandDispatcher<CommandSource?>) {
        // Thing to note, arguments are handled in alphabetical order.

        val clear = Commands.literal("reloadskin").executes { obj: CommandContext<CommandSource?>? -> run(obj) }

        dispatcher.register(clear)
    }

    private fun run(ctx: CommandContext<CommandSource?>?): Int {
        try {
            PacketHandler.sendToPlayer(ClientClearSkinCache, ctx!!.source!!.asPlayer())
            CustomSkinManager.sendAllToPlayer(ctx.source!!.asPlayer(), false)
        } catch (e: CommandSyntaxException) {
            println("This command can only be run from the console")
        }
        ctx!!.source!!.sendFeedback(TranslationTextComponent("messages.reloadSkinSuccess"), false)
        return Command.SINGLE_SUCCESS
    }
}